pkg/services/searchV2/allowed_actions.go

```diff
@@ -0,0 +1,257 @@
+package searchV2
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sort"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+func (s *StandardSearchService) addAllowedActionsField(ctx context.Context, orgId int64, user *models.SignedInUser, response *backend.DataResponse) error {
+	references, err := getEntityReferences(response)
+	if err != nil {
+		return err
+	}
+
+	allAllowedActions, err := s.createAllowedActions(ctx, orgId, user, references)
+	if err != nil {
+		return err
+	}
+
+	if len(response.Frames) == 0 {
+		return errors.New("empty response")
+	}
+
+	frame := response.Frames[0]
+
+	allowedActionsField := data.NewFieldFromFieldType(data.FieldTypeJSON, len(allAllowedActions))
+	allowedActionsField.Name = "allowed_actions"
+	frame.Fields = append(frame.Fields, allowedActionsField)
+
+	for i, actions := range allAllowedActions {
+		js, _ := json.Marshal(actions)
+		jsb := json.RawMessage(js)
+		allowedActionsField.Set(i, jsb)
+	}
+
+	return nil
+}
+
+type allowedActions struct {
+	EntityType entityKind `json:"type"`
```

Comment:
```suggestion
	EntityKind entityKind `json:"kind"` // eg: datasource
	EntityType string `json:"type,omitempty"` // eg: prometheus, influx
```

Reply:
renamed the `type` to `kind`, did not add a separate `type` prop. Do you think we need both? It wouldn't be trivial to add them since all we work with for figuring out the permissions are the `uid`s